janus/lib/server.rb

```diff
@@ -23,6 +23,8 @@ class Server < Etna::Server
 
     get '/project/:project_name', action: 'admin#project', auth: { user: { can_edit?: :project_name } }
 
+    get '/whoami', action: 'user#whoami', auth: { user: { active?: true } }
```

Comment:
Adding this route to Janus because we need to extract the user information from just the token in order to add them to the new project.

Reply:
Why not just parse the token?

Reply:
Would you need to pass around a config file (or database record or whatever) with the Janus public key and the hash algorithm to do that? So if either of those changes, we'd have to push those changes to the clients?

Reply:
Or, I guess that's an argument for keeping the commands on a polyphemus server and the installed etna gem just passes calls along to polyphemus. For the moment I was staying away from that workflow (and sticking to etna gem makes HTTP calls directly), since there currently isn't an API on top of polyphemus, and seems like for straightforward workflows, polyphemus would become an unnecessary potential point of failure.

Reply:
The public key is technically safe to pass along to our clients, although we'd need to include all possible public keys (prod, staging, dev) and users would need to update the gem when the key changed (for instance, revoking currently requires changing out the cert).

What about, instead of a new endpoint, we just -include- basic user info in say the login info if you are already authenticated?  I think it is reasonable to overload the login endpoint this way.

Reply:
The token is not encrypted, it is just base64-encoded. If you want you can just parse it using split & Base64.decode64. Timur et al parse data out of the token if you want an example.

Reply:
Oh, I didn't realize that...makes things a lot simpler. From looking at the code in `Etna::SignService`, I thought the token was encrypted. Thanks!

Reply:
This is a good primer on JWT: https://jwt.io/introduction/ - basically, there are three parts, base64 encoded - the first two are a JSON header and payload, the last is a signature of the first two parts made with the Janus (or whatever) private key, which may be verified with the public key. However, we may decode the payload without verifying it, the signature just tells us that Janus approved the payload.